### DISCUSSION TITLE
Event Return in an Edit inside a FocusScope

### DISCUSSION THREAD

**Discussion by Lianeker** [slint-ui/slint#5744] [Q&A]
I got this code segment in my appwindow.slint
```
FocusScope {
    height: 33%;
    key-pressed(event) => {
        debug(event.text);
        if (event.text == Key.Return) {
            Solve.button-pressed(input.text); 
            debug("ret");
        }
        if (event.text == Key.Tab) {
            Logic.button-pressed("Tab");
            debug("tab");
        }
        accept
    }
    input := LineEdit {
        height:100%;
        width:100%;
        font-size: 20px;
        text: "\{text_edit}";
        horizontal-alignment: right;
    }
}
```
The console registers "tab" when tab is pressed but not "ret" when return is pressed, "input" is forcefully focused always. All I need is to handle return input also, even from backend. Any tip would be appreciated.

**Comment by tronical** [comment]
The reason for not seeing "ret" is because the LineEdit unconditionally accepts the return key. So a "surrounding" focus scope serves a good purpose of handling keys that the children did /not/ handle, but it looks like you're looking for a way to intercept keys before the `LineEdit` handles them. We don't have an API for that yet :-(.
